### PR TITLE
Support subnet config

### PR DESCRIPTION
**Pull request by felipemadero**
Closes https://github.com/ava-labs/avalanche-network-runner/issues/348

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -356,13 +356,13 @@ curl -X POST -k http://localhost:8081/v1/control/createblockchains -d '{"pluginD
 avalanche-network-runner control create-blockchains '[{"vm_name":"'$VM_NAME'","genesis":"'$GENESIS_PATH'", "subnet_id": "'$SUBNET_ID'"}]' --plugin-dir $PLUGIN_DIR
 ```
 
-To create a blockchain with a subnet id, and both chain config and network upgrade file paths (requires network restart):
+To create a blockchain with a subnet id, and chain config, network upgrade, subnet config file paths (requires network restart):
```
Comment (holisticode) [inline README.md:359]: Nit: when there are multiple elements in a sentence, "and" is usually put before the last one.

File: cmd/control/control.go [hunk 2]
```diff
@@ -172,6 +173,12 @@ func newStartCommand() *cobra.Command {
 		"",
 		"[optional] JSON string of map from chain id to its upgrade file contents",
 	)
+	cmd.PersistentFlags().StringVar(
```
Comment (holisticode) [inline cmd/control/control.go:176]: Looks like the flags part and the map unmarshal part  are always the same? Maybe we could put them into functions?

File: rpcpb/rpc.proto [hunk 3]
```diff
@@ -331,6 +337,11 @@ message RestartNodeRequest {
   // If specified, will create a file "chainname/upgrade.json" with
   // the contents provided here.
   map<string, string> upgrade_configs = 5;
+
+  // Map of subnet id to subnet config file contents.
+  // If specified, will create a file "subnetid.json" under subnets config dir with
+  // the contents provided here.
+  map<string, string> subnet_configs = 6;
```
Comment (holisticode) [inline rpcpb/rpc.proto:344]: Shouldn't we actually set `optional` to all these configs elements?

File: rpcpb/rpc.proto [hunk 4]
```diff
@@ -359,6 +370,11 @@ message AddNodeRequest {
...
```
Comment (holisticode) [inline rpcpb/rpc.proto:377]: As well here

File: rpcpb/rpc.proto [hunk 5]
```diff
@@ -408,6 +424,7 @@ message LoadSnapshotRequest {
   map<string, string> upgrade_configs = 6;
   optional string global_node_config  = 7;
   optional bool reassign_ports_if_used = 8;
+  map<string, string> subnet_configs = 9;
```
Comment (holisticode) [inline rpcpb/rpc.proto:427]: ...and here?

**Review by holisticode — COMMENTED** [review]
All in all looks good to me.

Only what looks like minor issues commented.

File: rpcpb/rpc.proto (hunk 3 above)
Comment (felipemadero) [reply rpcpb/rpc.proto:344]: optional makes the value to be nil if not given, which is already the behaviour for maps

File: rpcpb/rpc.proto (hunk 4 above)
Comment (felipemadero) [reply rpcpb/rpc.proto:377]: same

File: rpcpb/rpc.proto (hunk 5 above)
Comment (felipemadero) [reply rpcpb/rpc.proto:427]: same

File: rpcpb/rpc.proto (hunk 3 above)
Comment (holisticode) [reply rpcpb/rpc.proto:344]: OK but to me it's makes things explicit for every one else. I did not know for example  that the value is nil if not given.
 
So I think it's clearer to set it optional where it needs to be optional

File: cmd/control/control.go (hunk 2 above)
Comment (felipemadero) [reply cmd/control/control.go:176]: Would rather prefer another PR with changes for all params https://github.com/ava-labs/avalanche-network-runner/issues/352

File: rpcpb/rpc.proto (hunk 3 above)
Comment (felipemadero) [reply rpcpb/rpc.proto:344]: also the change should be applied to all map params in RPC so another PR seems best

**Review by holisticode — APPROVED** [review]

**Pull request merged**